src/diff/props.js

```diff
@@ -103,49 +97,39 @@ export function setProperty(dom, name, value, oldValue, isSvg) {
 		} else {
 			dom.removeEventListener(name, proxy, useCapture);
 		}
-	} else if (
-		name !== 'list' &&
-		name !== 'tagName' &&
-		// HTMLButtonElement.form and HTMLInputElement.form are read-only but can be set using
-		// setAttribute
-		name !== 'form' &&
-		name !== 'type' &&
-		name !== 'size' &&
-		name !== 'download' &&
-		name !== 'href' &&
-		name !== 'contentEditable' &&
-		!isSvg &&
-		name in dom
-	) {
-		dom[name] = value == null ? '' : value;
-	} else if (typeof value != 'function' && name !== 'dangerouslySetInnerHTML') {
-		if (name !== (name = name.replace(/xlink:?/, ''))) {
-			if (value == null || value === false) {
-				dom.removeAttributeNS(
-					'http://www.w3.org/1999/xlink',
-					name.toLowerCase()
-				);
-			} else {
-				dom.setAttributeNS(
-					'http://www.w3.org/1999/xlink',
-					name.toLowerCase(),
-					value
-				);
-			}
+	} else if (name !== 'dangerouslySetInnerHTML') {
+		if (isSvg) {
+			// Normalize incorrect prop usage for SVG:
+			// - xlink:href / xlinkHref --> href (xlink:href was removed from SVG and isn't needed)
+			// - className --> class
+			name = name.replace(/xlink[H:h]/, 'h').replace(/Name$/, '');
 		} else if (
-			value == null ||
-			(value === false &&
-				// ARIA-attributes have a different notion of boolean values.
-				// The value `false` is different from the attribute not
-				// existing on the DOM, so we can't remove it. For non-boolean
-				// ARIA-attributes we could treat false as a removal, but the
-				// amount of exceptions would cost us too many bytes. On top of
-				// that other VDOM frameworks also always stringify `false`.
-				!/^ar/.test(name))
+			name !== 'href' &&
+			name !== 'list' &&
+			name !== 'form' &&
+			name !== 'download' &&
+			name in dom
 		) {
-			dom.removeAttribute(name);
-		} else {
+			try {
+				dom[name] = value == null ? '' : value;
+				// labelled break is 1b smaller here than a return statement (sorry)
+				break o;
+			} catch (e) {}
+		}
+
+		// ARIA-attributes have a different notion of boolean values.
+		// The value `false` is different from the attribute not
+		// existing on the DOM, so we can't remove it. For non-boolean
+		// ARIA-attributes we could treat false as a removal, but the
+		// amount of exceptions would cost us too many bytes. On top of
+		// that other VDOM frameworks also always stringify `false`.
+
+		if (typeof value === 'function') {
+			// never serialize functions as attribute values
+		} else if (value != null && (value !== false || /^ar/.test(name))) {
```

Comment:
This saves a couple bytes on my machine.

```suggestion
		} else if (
			value != null &&
			(value !== false || (name[0] === 'a' && name[1] === 'r'))
		) {
```

Reply:
dang! I played around with this and for some reason never thought of that.